zio-http/src/main/scala/zhttp/socket/SocketProtocol.scala

```diff
@@ -9,7 +14,8 @@ import zio.duration.Duration
 sealed trait SocketProtocol { self =>
```

Comment:
We can give additional type-safety and separate out server config from client config using a phantom type.

Reply:
Got it. I'll have to read more about phantom types for this.